l10n_ch_base_bank/models/invoice.py

```diff
@@ -67,7 +67,20 @@ def onchange_reference(self):
                 self._is_isr_reference()
             except exceptions.ValidationError:
                 return
-            self.reference_type = 'isr'
+            self.write({'reference_type': 'isr'})
+
+    def _get_bank_id(self, partner, journal_id, ref_type):
```

Comment:
Here we want to use the method to get partner either from the payment mode or on the invoice.

Reply:
I could replace it only for write method, create will use old one